### PR TITLE
Add a mock my-maps implementation to test covalent data-tables

### PR DESCRIPTION
**Pull request by agxs**
Created this to have a play with data tables from the Covalent library, since Angular Material 2 doesn't have them yet.

I recommend checking out the branch locally and running it. My Maps appears in the menu sidebar.

### CODE REVIEW

**Comment by marksmall** [comment]
It looks good as is, I guess the question is, can it handle links to select boxes (I'm thinking Data Download here), not sure by adding buttons it has proved it can also contain complex form elements, but on the other hand, what use would it be without that ability really

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
```
Comment (marksmall) [inline src/app/app.module.ts]: Are annotations used by you, or are than used by covalent? I can't see any use other than being imported into the app.module.

**Comment by marksmall** [comment]
Yeah, this looks good. Something I've noticed is that in a lot of my code, I'm referencing a Component in another components module directly e.g. in clients/geology.config.ts there are imports such as:

```javascript
import { MyMapsOpenComponent } from '../../menu/components/common/my-maps/my-maps-open.component';
```

This feels brittle to me if you have to rename either the location of the component or it's name. I've not tried, but it is likely the module the MyMapsOpenComponent belongs to is exporting the component, can we import it from it's module instead. This may make things a little less brittle.

File: src/app/app.module.ts (hunk 1 above)
Comment (agxs) [inline src/app/app.module.ts]: They're used by covalent. We can also put in a NoopAnimationsModule if the payload size is too big due to this, or we want no animations.

Specifically, it's the tool-tip that needs an animations module of some sort.

**Comment by agxs** [comment]
This is because the config file needs to reference which components it's putting in the sidebar panel, as they're created dynamically. The `<client>-config.ts` is the only file that should be doing this.

**Comment by agxs** [comment]
The links to select boxes thing is more a "feature" of ExtJS rather than because it's a desirable UI. The normal HTML `<select>/<option>` should be fine. Plus it can be backed by an Observable so that should make that code *so much* better.

**Pull request merged**